FGEGraphics/GraphicsHelpers/Models/ShapeGenerators.cs

```diff
@@ -0,0 +1,254 @@
+                if (i < stacks && j < slices)
+                {
+                    uint currentRow = i * (slices + 1);
+                    uint nextRow = (i + 1) * (slices + 1);
+                    indices[index++] = currentRow + j;
+                    indices[index++] = nextRow + j;
+                    indices[index++] = currentRow + j + 1;
+                    indices[index++] = currentRow + j + 1;
+                    indices[index++] = nextRow + j;
+                    indices[index++] = nextRow + j + 1;
+                }
+            }
+        }
+        return GetModelAfterGenerating(modelEngine, "sphere", vertices, normals, texCoords, indices);
+    }
+
+    /// <summary>Generates a 2D circle model.</summary>
+    public static Model Generated2DCircle(float radius, uint slices, ModelEngine modelEngine)
+    {
+        uint vertexCount = slices;
+        uint numIndices = slices * 3;
+        List<Vector3> vertices = new((int)vertexCount);
+        List<Vector3> normals = new((int)vertexCount);
+        List<Vector2> texCoords = new((int)vertexCount);
+        uint[] indices = new uint[numIndices];
+        int index = 0;
+        for (uint i = 0; i <= slices; i++)
+        {
+            float phi = i * 2 * MathHelper.Pi / slices;
+            float sinPhi = (float)Math.Sin(phi);
+            float cosPhi = (float)Math.Cos(phi);
+            float x = radius * cosPhi;
+            float y = radius * sinPhi;
+            float z = 0;
+            vertices.Add(new Vector3(x, y, z));
+            normals.Add(new Vector3(x, y, z).Normalized());
+            texCoords.Add(new Vector2(cosPhi, sinPhi));
+            if (i < slices)
+            {
+                indices[index++] = 0;
+                indices[index++] = i + 1;
+                indices[index++] = i + 2;
+            }
+        }
+        return GetModelAfterGenerating(modelEngine, "circle", vertices, normals, texCoords, indices);
+    }
+
+    /// <summary>Generates a 3D cylinder model.</summary>
+    public static Model GenerateCylinder(float radius, float height, uint slices, uint stacks, ModelEngine modelEngine)
+    {
+        uint vertexCount = (stacks + 1) * (slices + 1);
+        uint numIndices = (stacks * slices * 6) * 2;
+        List<Vector3> vertices = new((int)vertexCount);
+        List<Vector3> normals = new((int)vertexCount);
+        List<Vector2> texCoords = new((int)vertexCount);
+        uint[] indices = new uint[numIndices];
+        int index = 0;
+        for (uint i = 0; i <= slices; i++)
+        {
+            float phi = i * 2 * MathHelper.Pi / slices;
+            float sinPhi = (float)Math.Sin(phi);
+            float cosPhi = (float)Math.Cos(phi);
+            float x = radius * cosPhi;
+            float y = radius * sinPhi;
+            float z = height;
+            vertices.Add(new Vector3(x, y, z));
+            normals.Add(new Vector3(x, y, z).Normalized());
```

Comment:
this normal seems wrong. If this is the top/bottom circle code i'd expect a constant normal (circles are flat)